plugins/mana/parameters.go

```diff
@@ -31,4 +37,5 @@ func init() {
 	flag.Bool(CfgAllowedAccessFilterEnabled, false, "if filtering on access mana pledge nodes is enabled")
 	flag.Bool(CfgAllowedConsensusFilterEnabled, false, "if filtering on consensus mana pledge nodes is enabled")
 	flag.Bool(CfgManaEnableResearchVectors, false, "enable mana research vectors")
+	flag.Duration(CfgPruneConsensusEventLogsInterval, 5*time.Minute, "interval to check and prune consensus event storage")
```

Comment:
@acha-bill don't we stick to the counter based approach?

Reply:
what counter approach?

Reply:
Yes. Pruning x% of the events from storage when we reach max size.

Reply:
This is the interval to check the size and prune if necessary.

Reply:
@acha-bill if you only have the ticker there would be more events added in `tickerInterval` at a certain TPS than events that get removed. So log storage just grows. With current values, assuming best case scenario for tx input count, this would be 18 TPS. I'm not saying we should tweak the parameters, becasue this problem will arise at some TPS for sure becasue of the design. I'm saying to add the size check to `logPledge` and `logRevoke`, and trigger the pruning from there if the buffer gets full before the tick comes. Kind of the same flow as I suggested for the `csv` logger.

Reply:
There's no point of ticking if we check the size every single time we save an event. If the buffer is full, the logger will always know first.

It only makes sense to go with 1 approach. 
The choice of approach simply depends on the TPS.

I thought about this and went with the timer coz it's more configurable.

Reply:
Alright, let's go with the timer approach.
Notes:
 - Decrement `consensusEventsLogsStorageSize` when doing `DeleteEntriesFromStore`.
 - We could add an adaptive window approach to prevent the growing of the storage. Downside being is that when there is high tps, your window to reconstruct the past vector will decrease. (unless you set greater storage allowance). Let's discuss this in a call.

Reply:
>We could add an adaptive window approach to prevent the growing of the storage.

I think it's fine as is tbh. We can make more optimizations are we see fit later on.